core/metrics/src/main/scala/org/typelevel/otel4s/metrics/ObservableMeasurement.scala

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.metrics
+
+import org.typelevel.otel4s.Attribute
+
+trait ObservableMeasurement[F[_], A] {
+  def record(value: A, attributes: Attribute[_]*): F[Unit]
```

Comment:
@christiankjaer it would be nice to preserve laziness via macro. It should be nearly identical to `HistogramMacro`.
Could you take a look there too?

Reply:
Sure. The reason why I didn't look at the macro is that I kinda assumed that the observable stuff is for instrumenting from the outside, and the noop is then rarely needed. I will take a look.

Reply:
> observable stuff is for instrumenting from the outside

Since the callback is a function `ObservableMeasurement[F, A] => F[Unit]`, it will never be executed if we use a noop implementation. Therefore no need to worry about allocations.

So, we can keep it as-is and skip macro.